internal/server/routes.go

```diff
@@ -116,6 +116,9 @@ func (s *Server) GenerateRoutes(promRegistry prometheus.Registerer) Routes {
 
 	get(a, noAuthn, "/api/version", a.Version)
 
+	get(a, authn, "/api/settings", a.GetSettings)
```

Comment:
getting the settings need to be an unauthenticated endpoint, otherwise you can't get them to log in.

Reply:
I believe it only needs to be public for the first user during login (as this is the only time an un-authenticated user needs to set a new password, thus needing this endpoint unauthenticated). 

If all other cases are that user MUST be logged in to set a new password (hence authenticated), I thought we could handle the 'unauthenticated user' portion because they would always be the default settings (which only requires min length of 8). They would update these settings after they're setup, which by then, you no longer need to retrieve password settings unauthenticated (?).

Reply:
resolved in chat